libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -1466,6 +1466,27 @@ bool AP_Arming::serial_protocol_checks(bool display_failure)
     return true;
 }
 
+//Check for estop
+bool AP_Arming::estop_checks(bool display_failure)
+{
+    if (!SRV_Channels::get_emergency_stop()) {
+       // not emergency-stopped, so no prearm failure:
+       return true;
+    }
+    // vehicle is emergency-stopped; if this *appears* to have been done via switch then we do not fail prearms:
+    const RC_Channel *chan = rc().find_channel_for_option(RC_Channel::AUX_FUNC::ARM_EMERGENCY_STOP);
+    if (chan == nullptr) {
+        // RC channel not configured for arm_emergency_stop option, so estop not activated via this switch, prearm failure
+        return false;  //prearm failure
+    }
+    if (chan->get_aux_switch_pos() == RC_Channel::AuxSwitchPos::LOW) {
+         // switch configured and  is in estop position, so likely the reason we are estopped, so no prearm failure
+         return true;  // no prearm failure
+    }   
+    check_failed(display_failure,"Motors Emergency Stopped");
```

Comment:
Gah.  My formulation here means we won't get this string out in the case that chan is nullptr :-(

Reply:
I fixed that when I made the logic you suggested really work